docs/proposed/psa-driver-interface.md

```diff
@@ -536,6 +537,37 @@ The core will only call `acme_export_public_key` on a private key. Drivers imple
 
 The core guarantees that the size of the output buffer (`data_size`) is sufficient to export any key with the given attributes. The driver must set `*data_length` to the exact size of the exported key.
 
+### Builtin keys for opaque drivers
+
+An opaque driver can be pre-provisioned from the factory with keys. These keys are never seen by the PSA Crypto core, but in order to be able to use these keys, they need to be declared to the core.
+
+An opaque driver can declare such keys in its JSON descriptor file. The declaration is an array of strings, where each string is the name of a preprovisioned key, e.g.:
+```
+{
+    "builtin_keys": ["pre_key_1", "pre_key_2"]
+}
+```
+
+The key names need to be compatible with C language variable naming constraints, i.e. they cannot contain spaces and cannot start with a digit.
+
+For each declared key, the respective driver needs to define and externally declare the corresponding PSA Crypto key attributes (`psa_key_attributes_t`). The name of this variable, for each key, is `{prefix}_builtin_{keyname}_attributes`, where `{prefix}` is the implementing driver's prefix (lowercase), and `{keyname}` is the lowercase name of the key from the `builtin_keys` array.
+
+E.g. for a key called `"identity"`, a driver called `"acme"` would need to expose:
+```
+extern psa_core_key_attributes_t acme_builtin_identity_attributes;
+
+psa_key_attributes_t acme_builtin_identity_attributes = {
+    ...
+};
+```
+
+In order to use these keys, the JSON conversion process will generate a preprocessor symbol for each key, called `{PREFIX}_BUILTIN_{KEYNAME}_ID`, where `{PREFIX}` is the uppercase driver prefix, and `{KEYNAME}` the uppercase name of the key. This symbol yields a `psa_key_id_t` from the PSA key ID vendor range, and can be supplied as an argument to `psa_open_key()` to get a handle to access the builtin key.
+
+For the same `"identity"` key, this would cause the user to be able to get a handle for that key with the following call:
+```
+status = psa_open_key(ACME_BUILTIN_IDENTITY_ID, &handle);
```

Comment:
`psa_open_key` no longer exists in the API specification. It still exists in Mbed TLS, but [not for much longer](https://github.com/ARMmbed/mbedtls/pull/3547) except as a mostly-no-op for backward compatibility.

Reply:
But for now, it's there, so I thought I could reference it to make the spec easier to understand.

Of course, after openless is in place, we'll be able to use the generated IDs directly.